x-pack/solutions/observability/plugins/apm/public/components/app/metrics/static_dashboard/dashboards/get_dashboard_file_name.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getSdkNameAndLanguage, getIngestionPath } from '@kbn/elastic-agent-utils';
+
+interface DashboardFileNamePartsProps {
+  agentName: string;
```

Comment:
can `AgentName` type be used here?
```suggestion
  agentName: AgentName;
```

Reply:
I need to change the type from the API response [here](https://github.com/elastic/kibana/pull/211822/files/aafbb80b7b3cfe9afc842e8c69b8471f555697f7#diff-81c749c7ca9993b80335cc4ba338f5b08508210ed20f487aa132cbcd7af71640R26-R27) (which is set to string so I just use the same type here, I don't see a reason to set it to string there as in theory we should return type AgentName - maybe there were some edge cases when the type don't match when it's something custom, also we need the EDOT types added [here](https://github.com/elastic/kibana/pull/211468/files#diff-4002324d26c449874154dc9eb897841d41d41b771a5f1387f9598582b0febc7bR23) in the future when we have EDOT dashboards) so I will check if that can be done everywhere and change it

Reply:
Yeah, let's see how this evolves. Then we can create a tech debt issue. Thanks